client/packages/invoices/src/OutboundShipment/DetailView/OutboundLineEdit/OutboundLineEdit.tsx

```diff
@@ -66,13 +82,45 @@ export const OutboundLineEdit: React.FC<ItemDetailsModalProps> = ({
   const { next, disabled: nextDisabled } = useNextItem(currentItem?.id);
   const { isDirty, setIsDirty } = useDirtyCheck();
   const height = useKeyboardHeightAdjustment(700);
+  const { warning } = useNotification();
+  useFocusNumberOfPacksInput(draft);
+
   const placeholder = draftOutboundLines?.find(
     ({ type, numberOfPacks }) =>
       type === InvoiceLineNodeType.UnallocatedStock && numberOfPacks !== 0
   );
 
+  const onSave = async () => {
+    if (!isDirty) return;
+
+    await mutateAsync(draftOutboundLines);
+
+    if (!draft) return;
+
+    const { barcode } = draft;
+    const barcodeExists = !!barcode?.id;
+    if (!barcode || !currentItem || barcodeExists) return;
+
+    draftOutboundLines
+      .filter(line => line.numberOfPacks > 0)
```

Comment:
I think it makes more sense to also filter by batch, even though they are disabled, but that's hard to know by looking at the code here

Reply:
If there are more then two counted rows (if batch is not set or if there are two stock lines with the same batch, I think we would be adding it two times ?

Reply:
I'm not following what you mean. The barcode is only inserted if there is no existing barcode, so there won't be any lines disabled. The batch might not be part of the scanned barcode too - depends on the format of the barcode whether that's provided or not.

Reply:
From my understanding by reading the code, if we scan a barcode with batch or with no batch, that barcode doesn't exist, so we get to select and item, an item is selected and rows are enabled that either match the batch (or all rows are selected if batch is not matched at all), now if we count multiple rows, we would insert the barcode multiple times. We could count rows with the same pack size or different pack size or different batch ?

Reply:
ah. gotcha, thanks - good catch!